sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/exporter/SpanExporter.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace
+package exporter
+
+import cats.Applicative
+import cats.Monoid
+import cats.Parallel
+import cats.data.NonEmptyList
+import cats.syntax.foldable._
+import cats.syntax.parallel._
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+
+/** An interface that allows different tracing services to export recorded data
+  * for sampled spans in their own format.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/sdk/#span-exporter]]
+  *
+  * @tparam F
+  *   the higher-kinded type of a polymorphic effect
+  */
+trait SpanExporter[F[_]] {
+
+  /** Called to export sampled [[data.SpanData SpanData]].
+    *
+    * '''Note''': the export operations can be performed simultaneously
+    * depending on the type of span processor being used. However, the batch
+    * span exporter will ensure that only one export can occur at a time.
```

Comment:
Sorry, I'm confused by this comment. What is the "batch span exporter"?

Reply:
Oh, it should be `batch span processor` instead. Once I add it, I will use a scaladoc reference. 

The implementation will be based on:
https://github.com/typelevel/otel4s/pull/325/files#diff-ff368903b03e16d45e93659be2898e27f243b1a48dd4af8e516e9ee69048dbba

Reply:
I updated scaladoc